client/components/overviewWidget/StyleSelector.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import Styles from './Styles.jsx';
+import axios from 'axios';
+
+class StyleSelector extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      product: null,
+      styles: [],
+      selectedStyle: ''
+    };
+  }
+
+  componentDidMount() {}
+
+  componentDidUpdate(prevProps) {
+    if (this.props.product !== prevProps.product) {
+      axios.get(`/api/fec2/hrnyc/products/${this.props.product.id}/styles`)
```

Comment:
Am I wrong or did you also fetch this same data in another component? If that is the case any reason you can't move the styles data up a level and then pass it down as props?